internal/ignition/ignition.go

```diff
@@ -1513,3 +1524,67 @@ func proxySettingsForIgnition(httpProxy, httpsProxy, noProxy string) (string, er
 	}
 	return buf.String(), nil
 }
+
+func getIcspFileFromInstallConfig(cfg []byte) (string, error) {
+	contents, err := getIcsp(cfg)
+	if err != nil {
+		return "", err
+	}
+	if contents == nil {
+		return "", nil
+	}
+
+	icspFile, err := ioutil.TempFile("", "icsp-file")
+	if err != nil {
+		return "", err
+	}
+	if _, err := icspFile.Write(contents); err != nil {
+		os.Remove(icspFile.Name())
+		return "", err
+	}
+	icspFile.Close()
```

Comment:
Need to defer this prior to line 1541, otherwise we will leak a file descriptor if the Write() fails.

Reply:
Changed to add close if Write() fails